### PR TITLE
gh #46 Remove ODM specific apis - phase1

### PR DESCRIPTION
**Pull request by arjunbinu**
L1 and L2 PR
https://github.com/rdkcentral/rdkv-halif-test-tvsettings/pull/57

**Main Objective**
The current activity focuses on removing soon-to-be deprecated ODM APIs and types from the TvSettings HAL headers, particularly those exposed to plugins and factory code.

**APIs Removed**
From include/tvSettingsExtODM.h:
int ReadCapabilitiesFromConfODM(...)
Defintion moved to the plugin.

_From include/tvSettingsODM.h:_
tvError_t **GetSupportedVideoFormatsODM**(unsigned int *videoFormats, unsigned short *numberOfFormats)
tvVideoHDRFormat_t **GetCurrentVideoFormatODM**(void)
tvError_t **GetTVSupportedDimmingModesODM**(char **dimmingModes, unsigned short *numDimmingModes)
Stubbed replacements are now present in tvSettings.h.
Conversion APIs like **ConvertVideoFormatToHDRFormatODM** and **ConvertHDRFormatToContentFormatODM**
Removed due to the deprecation of types like tvhdr_type_t and tvVideoHDRFormat_t.
int **GetPanelIDODM**(char *panelid)
Moved definition to the plugin. Further updates will encapsulate Panel ID retrieval in Phase 2.
tvError_t **TvSyncCalibrationInfoODM**()
Removed as the calibration data will now be handled differently.
tvError_t **GetTVSupportedDolbyVisionModesODM**(pic_modes_t *dvModes[],unsigned short *count)
tvError_t **SetWakeupConfig**(const tvWakeupSrcType_t src_type, const bool value);
tvError_t **SetTVDolbyVisionModeODM**(const char * dolbyMode);
int **GetCMSDefault**(tvComponentType_t component_type);

Note:
Additional function removals are included in preparation for Phase 2.

**Modified APIs**
_In include/tvSettings.h:_
tvError_t GetCurrentVideoSource(int *currentSource)
Updated to: tvError_t GetCurrentVideoSource(tvVideoSrcType_t *currentSource)
Rationale: To align with the expected type, ensuring consistency in API usage.

These changes are foundational for the overall API clean-up and ensure the TvSettings HAL interfaces are maintainable, and aligned with the upcoming architectural requirements.

### CODE REVIEW

**Review by lucienkennedylamb — APPROVED** [review]
Looks OK.

File: include/tvSettings.h [hunk 1]
```diff
@@ -283,7 +283,7 @@ tvError_t GetCurrentVideoFrameRate(tvVideoFrameRate_t *format);
  *
  * @pre TvInit() should be called before calling this API
  */
-tvError_t GetCurrentVideoSource(int *currentSource);
+tvError_t GetCurrentVideoSource(tvVideoSrcType_t *currentSource);
```
Comment (Anbukannadhasan) [inline include/tvSettings.h:287]: APIs Removed
From include/tvSettings.h:
tvError_t GetTVSupportedHDR10Modes(tvDolbyMode_t *HDR10Modes[], unsigned short *count)
tvError_t GetTVSupportedHLGModes(tvDolbyMode_t *HLGModes[], unsigned short *count)
Replaced by a common API: GetTVSupportedDolbyVisionModes to retrieve supported modes for DolbyVision, HDR10, and HLG.

Are the above APIs are expected to replace in this phase 1?

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: include/tvSettingsODM.h [hunk 2]
```diff
@@ -56,92 +56,6 @@ extern "C"
...
 #endif
 
-/**
- * @brief Gets supported video formats of the system. To be deprecated soon.
- * @note SOC vendors can stub these ODM functions and mark it as weak
- *
- * This function returns all the supported content formats. 
- *
- * @param[out] videoFormats    - Bitwise ORed values of enum ::tvVideoHDRFormat_t
- * @param[out] numberOfFormats - Count of supported video formats. Min should be 1
```
Comment (hari22yuva) [inline include/tvSettingsODM.h:66]: Please check these structure / enum not used if not remove. Please check other also
tvhdr_type_t
tvWakeupSrcType_t,
tvVideoHDRFormat_t

File: include/tvSettings.h [hunk 3]
```diff
@@ -283,7 +283,7 @@ tvError_t GetCurrentVideoFrameRate(tvVideoFrameRate_t *format);
...
```
Comment (hari22yuva) [inline include/tvSettings.h:286]: please check required to update the test cases also?

**Review by hari22yuva — COMMENTED** [review]
Please check the review comments

File: include/tvSettingsODM.h [hunk 4]
```diff
@@ -235,134 +149,6 @@ tvError_t GetWBctrl(char *inputSrc, char *colortemp, char *color, char *ctrl, in
...
- *
- * @return int
- *
- * @retval 15                       - Luma default value
- * @retval 50                       - Hue or saturation default value
- * @retval -1                       - Input parameter is invalid
- *
- * @pre tvInit() should be called before calling this API
- */
- int GetCMSDefault(tvComponentType_t component_type);
```
Comment (Anbukannadhasan) [inline include/tvSettingsODM.h:254]: Are these expected to be remove? The API is not listed in the PR description.

File: include/tvSettingsODM.h [hunk 5]
```diff
@@ -235,134 +149,6 @@ tvError_t GetWBctrl(char *inputSrc, char *colortemp, char *color, char *ctrl, in
...
- * @retval tvERROR_INVALID_PARAM             - Input parameter is invalid
- * @retval tvERROR_INVALID_STATE             - Interface is not initialized
- * @retval tvERROR_OPERATION_NOT_SUPPORTED   - Operation is not supported
- * @retval tvERROR_GENERAL                   - Underlying failures - SoC, memory, etc
- *
- * @pre tvInit() should be called before calling this API
- *
- * @see GetTVDolbyVisionMode()
- */
-tvError_t SetTVDolbyVisionModeODM(const char * dolbyMode);
```
Comment (Anbukannadhasan) [inline include/tvSettingsODM.h:364]: SetTVDolbyVisionModeODM is expected to remove? This API is also not listed in the PR description

File: include/tvSettingsODM.h [hunk 6]
```diff
@@ -460,44 +246,6 @@ tvError_t GetTVSupportedHDR10ModesODM(pic_modes_t *dvModes[],unsigned short *cou
...
- * @return tvError_t
- *
- * @retval tvERROR_NONE                      - Success
- * @retval tvERROR_INVALID_PARAM             - Input parameter is invalid
- * @retval tvERROR_INVALID_STATE             - Interface is not initialized
- * @retval tvERROR_GENERAL                   - Underlying failures - SoC, memory, write failure, etc
- *
- * @pre TvInit() should be called before calling this API
-*/
-tvError_t SetWakeupConfig(const tvWakeupSrcType_t src_type, const bool value);
```
Comment (Anbukannadhasan) [inline include/tvSettingsODM.h:499]: SetWakeupConfig is not in PR description? but removed. Is it expected to be removed?

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: include/tvSettingsODM.h (hunk 6 above)
Comment (arjunbinu) [reply include/tvSettingsODM.h:499]: Yes, it is expected to be removed. It was missing from the PR description, but I have now added it.

File: include/tvSettingsODM.h (hunk 5 above)
Comment (arjunbinu) [reply include/tvSettingsODM.h:364]: Yes, it is expected to be removed. It was missing from the PR description, but I have now added it.

File: include/tvSettingsODM.h (hunk 4 above)
Comment (arjunbinu) [reply include/tvSettingsODM.h:254]: Yes, it is expected to be removed. It was missing from the PR description, but I have now added it.

**Review by hari22yuva — APPROVED** [review]
looks ok

**Pull request merged**